### ISSUE TITLE
Quick add workflow

### ISSUE THREAD

**Issue by frafra** [geopaparazzi/smash#149]
This is the current workflow to add a custom note, starting from the map:
1. Press on the add button (document icon with the "+")
2. Click on the form
3. Open the section
4. Insert data
5. Press back
6. Repeat from 2 to 4 for each relevant section
7. Press back at the end to save
8. Repeat from 1 to 7 for each note of the same type that needs to be added 

This is the proposed workflow:
1. Press on the add button (document icon with the "+")
2. Insert data
3. Press next section
4. Repeat from 2 to 4 for section
5. Press "Add new"
6. Repeat from 2 to 5 for each node of the same type that needs to be added, or press on "save" for the last element

That kind of workflow would reduce the number of buttons to be pressed on the screen when many notes should be added sequentially.

Potential issue: should the same location be used for all the data?

**Comment by frafra** [comment]
I think that the same location should be used for all the data if the user does not exit.

It would be great to have a way to add a new data, based on what has been just inserted. That would be useful when adding many similar notes.

**Comment by moovida** [comment]
@frafra , I am not sure how you imagine to pass from 1 to 2. You want to directly insert data, but you are assuming there is only a single form, right? It would not make sense in the case of surveys with multiple forms of choice.

**Comment by frafra** [comment]
> @frafra , I am not sure how you imagine to pass from 1 to 2. You want to directly insert data, but you are assuming there is only a single form, right? It would not make sense in the case of surveys with multiple forms of choice.

You are right, it could be a simplification when there is only one form specified, but SMASH still has a switch between manual positioning and GPS, so such intermediate menu cannot be removed easily, I guess.

**Comment by moovida** [comment]
Outch, you are right.

So that issue is "reduced" to precompilation of forms?

I am not sure about the add-new from within the form. That would mean that you do not see on the mao where you are adding the niew form, right?

**Comment by moovida** [comment]
I am not sure I understand the not changing position thing. Do you have a usecase in which you have to add multiple notes of the same type in the same position?

Regarding the prefill, I think it could work nocely if some unique it would be implemented for the form ( which would be necessary to solve #129 )

**Comment by frafra** [comment]
> I am not sure I understand the not changing position thing. Do you have a usecase in which you have to add multiple notes of the same type in the same position?

That is correct. You could have multiple notes in the exact same position (even of different types, potentially, but I did not though about that).

Prefill would need to link together forms somehow, I agree.
